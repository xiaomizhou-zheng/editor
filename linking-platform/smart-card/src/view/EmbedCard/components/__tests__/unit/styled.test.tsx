import React from 'react';
import { mount } from 'enzyme';
import { matchers } from 'jest-emotion';
import {
  Wrapper,
  LinkWrapper,
  IconWrapper,
  TextWrapper,
  Content,
} from '../../styled';

expect.extend(matchers);

describe('Wrapper', () => {
  it('should render with min-width when there is a minWidth', () => {
    const element = mount(<Wrapper minWidth={100} />);
    expect(element).toHaveStyleRule('min-width', '100px');
  });

  it('should not render with min-width when there is no minWidth', () => {
    const element = mount(<Wrapper />);
    expect(element).not.toHaveStyleRule('min-width', '');
  });

  it('should render with max-width when there is a maxWidth', () => {
    const element = mount(<Wrapper maxWidth={100} />);
    expect(element).toHaveStyleRule('max-width', '100px');
  });

  it('should not render with max-width when there is no maxWidth', () => {
    const element = mount(<Wrapper />);
    expect(element).not.toHaveStyleRule('max-width', '');
  });

  it('should have hover styles when isInteractive=true', () => {
    const element = mount(<Wrapper isInteractive={true} />);
    expect(element).toMatchSnapshot();
  });

  it('should not have hover styles when isInteractive=false', () => {
    const element = mount(<Wrapper />);
    expect(element).toMatchSnapshot();
  });
});

describe('LinkWrapper', () => {
  it('should render with minWidth when there is a minWidth', () => {
    const element = mount(<LinkWrapper minWidth={100} />);
    expect(element).toHaveStyleRule('min-width', '100px');
  });

  it('should not render with minWidth when there is no minWidth', () => {
    const element = mount(<LinkWrapper />);
    expect(element).not.toHaveStyleRule('min-width', '');
  });

  it('should render with minWidth when there is a minWidth', () => {
    const element = mount(<LinkWrapper maxWidth={100} />);
    expect(element).toHaveStyleRule('max-width', '100px');
  });

  it('should not render with minWidth when there is no minWidth', () => {
    const element = mount(<LinkWrapper />);
    expect(element).not.toHaveStyleRule('max-width', '');
  });

  it('should have hover styles when isInteractive=true', () => {
    const element = mount(<LinkWrapper isInteractive={true} />);
    expect(element).toMatchSnapshot();
  });

  it('should not have hover styles when isInteractive=false', () => {
    const element = mount(<LinkWrapper isInteractive={false} />);
    expect(element).toMatchSnapshot();
  });
});

describe('Content', () => {
  it('should not allow overflow content to be visible (iframe contents should handle scrolling)', () => {
    const element = mount(<Content isInteractive={false} />);
    expect(element).toHaveStyleRule('overflow', 'hidden');
  });

  it('should not allow overflow content to be visible (on hover as well)', () => {
    const element = mount(<Content isInteractive={true} />);
    expect(element).toHaveStyleRule('overflow', 'hidden');
  });
});

describe('IconWrapper', () => {
  it('should look like a placeholder when isPlaceholder=true', () => {
    const element = mount(<IconWrapper isPlaceholder={true} />);
    expect(element).toHaveStyleRule(
      'background-color',
      'var(--ds-skeleton,#EBECF0)',
    );
  });

  it('should look like a placeholder when isPlaceholder=false', () => {
    const element = mount(<IconWrapper isPlaceholder={false} />);
    expect(element).not.toHaveStyleRule(
      'background-color',
      'var(--ds-skeleton,#EBECF0)',
    );
  });
});

describe('TextWrapper', () => {
  it('should look like a placeholder when isPlaceholder=true', () => {
    const element = mount(<TextWrapper isPlaceholder={true} />);
    expect(element).toHaveStyleRule(
      'background-color',
      'var(--ds-skeleton,#EBECF0)',
    );
  });

  it('should look like a placeholder when isPlaceholder=false', () => {
    const element = mount(<TextWrapper isPlaceholder={false} />);
    expect(element).not.toHaveStyleRule(
      'background-color',
      'var(--ds-skeleton,#EBECF0)',
    );
  });
});
