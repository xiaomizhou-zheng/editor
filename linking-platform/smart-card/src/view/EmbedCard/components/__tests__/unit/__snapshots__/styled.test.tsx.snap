// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkWrapper should have hover styles when isInteractive=true 1`] = `
.emotion-0 {
  border-radius: 3px;
  margin: 0 auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: initial;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  position: relative;
  height: 432px;
}

.emotion-0:hover {
  background-color: var(--ds-background-neutral-subtle,#EBECF0);
}

.emotion-0:hover:after {
  background: var(--ds-background-neutral,#EBECF0) !important;
}

.emotion-0:hover .embed-header {
  opacity: 1;
}

.emotion-0:active {
  background-color: var(--ds-background-selected,#DEEBFF);
}

.emotion-0:after {
  content: '';
  background: transparent;
  -webkit-transition: background 0.3s,box-shadow 0.3s;
  transition: background 0.3s,box-shadow 0.3s;
  position: absolute;
  width: calc(100% + 16px);
  height: calc(100% + 8px);
  left: -8px;
  border-radius: 3px;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Styled(div)
  isInteractive={true}
>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;

exports[`LinkWrapper should not have hover styles when isInteractive=false 1`] = `
.emotion-0 {
  border-radius: 3px;
  margin: 0 auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: initial;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  position: relative;
  height: 432px;
}

.emotion-0:after {
  content: '';
  background: transparent;
  -webkit-transition: background 0.3s,box-shadow 0.3s;
  transition: background 0.3s,box-shadow 0.3s;
  position: absolute;
  width: calc(100% + 16px);
  height: calc(100% + 8px);
  left: -8px;
  border-radius: 3px;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<Styled(div)
  isInteractive={false}
>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;

exports[`Wrapper should have hover styles when isInteractive=true 1`] = `
.emotion-0 {
  border-radius: 3px;
  margin: 0 auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: initial;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  position: relative;
  height: 432px;
  margin-top: 10px;
}

.emotion-0:hover {
  background-color: var(--ds-background-neutral-subtle,#EBECF0);
}

.emotion-0:hover:after {
  background: var(--ds-background-neutral,#EBECF0) !important;
}

.emotion-0:hover .embed-header {
  opacity: 1;
}

.emotion-0:active {
  background-color: var(--ds-background-selected,#DEEBFF);
}

.emotion-0:after {
  content: '';
  background: transparent;
  -webkit-transition: background 0.3s,box-shadow 0.3s;
  transition: background 0.3s,box-shadow 0.3s;
  position: absolute;
  width: calc(100% + 16px);
  height: calc(100% + 8px);
  left: -8px;
  border-radius: 3px;
}

<Styled(div)
  isInteractive={true}
>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;

exports[`Wrapper should not have hover styles when isInteractive=false 1`] = `
.emotion-0 {
  border-radius: 3px;
  margin: 0 auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: initial;
  -webkit-transition: background 0.3s;
  transition: background 0.3s;
  position: relative;
  height: 432px;
  margin-top: 10px;
}

.emotion-0:after {
  content: '';
  background: transparent;
  -webkit-transition: background 0.3s,box-shadow 0.3s;
  transition: background 0.3s,box-shadow 0.3s;
  position: absolute;
  width: calc(100% + 16px);
  height: calc(100% + 8px);
  left: -8px;
  border-radius: 3px;
}

<Styled(div)>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;
